scripts/lobby.gd

```diff
@@ -55,18 +68,29 @@ func _on_ws_received_message(_msg_type: String) -> void:
 			DSGMessageType.PLAYER_JOINED:
 				self._player_joined(msg["payload"])
 			DSGMessageType.SERVER_START_GAME:
-				GlobalVariables.map = Utils.parse_dict_to_map(msg["payload"])
-				get_tree().change_scene_to_packed(self._board_scene)
+				self._server_start_game(msg["payload"])
 
 func _server_hello(payload: Variant):
 	if not payload["is_host"]:
 		self._setup_non_host()
-	GlobalVariables.player_id = payload["player_id"]
+	var player = payload["player"]
+	GlobalVariables.player_id = player["id"]
+	GlobalVariables.player_number = player["number"]
+	GlobalVariables.session_id = payload["session_id"]
+	self._info.text = "Welcome! You play %s." % GlobalVariables.COLOR_MAPPING[GlobalVariables.player_number]
```

Comment:
The `server_hello` message now also contains the list of all players that are already in the lobby.

See:
https://github.com/dsg-jam/ld51-server/blob/ad72f6c3a7a7d7507486315609c0f5ced8f6c18c/ld51_server/protocol/lobby.py#L16

We should at least use this to update the `amount_of_players`.

Reply:
18a9ee2